src/v/cluster/tm_stm.cc

```diff
@@ -130,6 +130,17 @@ tm_stm::do_get_tx(const kafka::transactional_id& tx_id) {
     return std::nullopt;
 }
 
+std::optional<tm_stm::txes_type::iterator>
```

Comment:
imho it's better to introduce the correct code in the first place than fixing it later, why did you decide to split?